### PR TITLE
Allow dashes in user urn

### PR DESCRIPTION
**Pull request by ben5448**
fix: allow dashes in user urn

Dashes are accepted in user urns for the CorpUserInfo documents. In the MAE
consumer for ownership facets, the owner urn is validated using a regex that
only accepts alphanumeric characters and underscores. If the owner's urn
contains a dash, that update triggers an error.

This fix: relax regex validation to allow a dash as a valid character in the urn.
Unit tests changes match verify the new filter and verify the filter allows
multiple dashes or underscores. No attempt is made for force the first or
last characters to be alphanumeric or to allow other characters.

CLOSES: User urn does not handle dashes consistently #1554

BREAKING CHANGES: None. This change relaxes a constraint so all existing
code should continue to work the same.

### CODE REVIEW

File: metadata-builders/src/test/java/com/linkedin/metadata/builders/search/BuilderUtilsTest.java [hunk 1]
```diff
@@ -24,9 +24,11 @@ public void testGetCorpUserOwners() {
     Owner owner1 = new Owner().setOwner(new CorpuserUrn("t1"));
     Owner owner2 = new Owner().setOwner(new CorpuserUrn("t2"));
     Owner owner3 = new Owner().setOwner(new CorpGroupUrn("t3"));
-    List<Owner> owners = Arrays.asList(owner1, owner2, owner3);
+    Owner owner4 = new Owner().setOwner(new CorpGroupUrn("t-4-t"));
```
Comment (keremsahin1) [inline metadata-builders/src/test/java/com/linkedin/metadata/builders/search/BuilderUtilsTest.java]: Maybe change this to `CorpuserUrn` to test both urn types?

File: metadata-builders/src/test/java/com/linkedin/metadata/builders/search/BuilderUtilsTest.java [hunk 2]
```diff
@@ -24,9 +24,11 @@ public void testGetCorpUserOwners() {
...
+    Owner owner5 = new Owner().setOwner(new CorpGroupUrn("t_5_t"));
+    List<Owner> owners = Arrays.asList(owner1, owner2, owner3, owner4, owner5);
     ownership.setOwners(new OwnerArray(owners));
-    assertEquals(BuilderUtils.getCorpUserOwners(ownership), Arrays.asList("t1", "t2"));
+    assertEquals(BuilderUtils.getCorpUserOwners(ownership), Arrays.asList("t1", "t2", "t3", "t-4-t", "t_5_t"));
```
Comment (keremsahin1) [inline metadata-builders/src/test/java/com/linkedin/metadata/builders/search/BuilderUtilsTest.java]: `getCorpUserOwners` should only return owners which are users not groups. I'm really curious how this test passed 😮

Comment (ben5448) [reply metadata-builders/src/test/java/com/linkedin/metadata/builders/search/BuilderUtilsTest.java]: Looking through the logs, I can't tell that these tests are getting executed.

File: metadata-builders/src/test/java/com/linkedin/metadata/builders/search/BuilderUtilsTest.java (hunk 1 above)
Comment (ben5448) [reply metadata-builders/src/test/java/com/linkedin/metadata/builders/search/BuilderUtilsTest.java]: I added tests or urn and groups. Adjusted the results to match what should be the correct values. These tests do not appear to run when building the project though. All the unit tests report success regardless of the data entered here. The tests are built, just not run.

**Comment by keremsahin1** [comment]
@ben5448 Thanks for this PR. It helped me figure out an underlying issue with some modules not being tested at all. I'll push the fix soon. I will just accept this now.

**Pull request merged**